File: 1272-remove-interval/1272-remove-interval.cpp
class Solution {
public:
    vector<vector<int>> removeInterval(vector<vector<int>>& intervals, vector<int>& toBeRemoved) {
        vector<vector<int>> ans;
        
        for(const auto& interval: intervals){
            if(toBeRemoved[1] <= interval[0] or interval[1] <= toBeRemoved[0]){
                // no overlap
                ans.push_back(interval);
            } else {
                // check if left part of the interval can make it
                
                if(interval[0] < toBeRemoved[0]){
                    ans.push_back({interval[0], toBeRemoved[0]});
                }
                
                //check if there is a right part to be added
                
                if(interval[1] > toBeRemoved[1]){
                    ans.push_back({toBeRemoved[1], interval[1]});
                }
            }
        }
        
        return ans;
    }
};

File: 1272-remove-interval/1272-remove-interval.py
class Solution:
    def removeInterval(self, intervals: List[List[int]], toBeRemoved: List[int]) -> List[List[int]]:
        
        
        ans = []
        
        for left, right in intervals:
            print("toBeRemoved:", toBeRemoved)
            print("left, right:", left, right)
            if toBeRemoved[1] <= left or right <= toBeRemoved[0]:
                print("no overlap")
                ans.append([left, right])
            else:
                if left < toBeRemoved[0]:
                    print("left hang")
                    ans.append([left, toBeRemoved[0]])
                if right > toBeRemoved[1]:
                    print("right hang")
                    ans.append([toBeRemoved[1], right])
        
        return ans
            